app/tests/integration/nimbus/test_e2e_create_experiment.py

```diff
@@ -126,3 +129,95 @@ def test_create_new_experiment_remote_settings(selenium, base_url):
+
+
+def test_create_new_experiment_remote_settings_reject(selenium, base_url):
+    experiment_name = f"name here remote {random.randint(0, 1000)}"
+
+    selenium.get(base_url)
+    home = HomePage(selenium, base_url).wait_for_page_to_load()
+    home.tabs[-1].click()  # Click drafts
+    current_experiments = len(home.tables[0].experiments)
+    experiment = home.create_new_button()
+    experiment.public_name = experiment_name
+    experiment.hypothesis = "smart stuff here"
+    experiment.application = "DESKTOP"
+
+    # Fill Overview Page
+    overview = experiment.save_and_continue()
+    overview.public_description = "description stuff"
+    overview.select_risk_brand_false()
+    overview.select_risk_revenue_false()
+    overview.select_risk_partner_false()
+
+    # Fill Branches page
+    branches = overview.save_and_continue()
+    branches.remove_branch()
+    branches.reference_branch_name = "name 1"
+    branches.reference_branch_description = "a nice experiment"
+    branches.feature_config = "No Feature Firefox Desktop"
+
+    # Fill Metrics page
+    metrics = branches.save_and_continue()
+
+    # Fill Audience page
+    audience = metrics.save_and_continue()
+    audience.channel = "Nightly"
+    audience.min_version = 80
+    audience.targeting = "US_ONLY"
+    audience.percentage = 50.0
+    audience.expected_clients = 50
+    audience.save_btn()
+    review = audience.save_and_continue()
+
+    # Review and approve
+    selenium.find_element_by_css_selector("#PageRequestReview")
+    review.launch_without_preview.click()
+    review.request_review.click_launch_checkboxes()
+    review.request_review.request_launch_button.click()
+    review.approve()
+    selenium.get("http://kinto:8888/v1/admin")
+    kinto_login = Login(selenium, base_url).wait_for_page_to_load()
+    kinto_login.kinto_auth.click()
+    kinto_dashbard = kinto_login.login()
+    bucket = kinto_dashbard.buckets[-1]
+    for item in bucket.bucket_category:
+        if "nimbus-desktop-experiments" in item.text:
+            item.click()
+            break
+    record = kinto_dashbard.record
+    record.action("reject")
+    kinto_dashbard = Dashboard(selenium, base_url)
+    modal = kinto_dashbard.reject_modal
+    modal.decline_changes()
+
+    # Load home page and wait for experiment to show in the Drafts tab
+    selenium.get(base_url)
+    for attempt in range(45):
+        try:
+            home = HomePage(selenium, base_url).wait_for_page_to_load()
+            home.tabs[-1].click()
+            new_experiments = len(home.tables[0].experiments)
+            assert new_experiments != current_experiments
+        except AssertionError:
+            time.sleep(2)
+            selenium.refresh()
+        else:
+            home.tabs[-1].click()
+            draft_experiments = home.tables[0]
+            for item in draft_experiments.experiments:
+                if experiment_name in item.text:
+                    item.click()
+                    break
```

Comment:
Just to confirm I understand this correctly&hellip;

So, we loop <45 times, and if the home page loads, we click on the last tab, get the length of the first table's experiments, and assert that the numbers are different (so an experiment was added). If our assertion fails, we fall into the `except` block, sleep for 2 seconds, and refresh the browser (1). Now, in the `else` flow, we click on the last tab (drafts) and assert that our experiment name is visible (and if so, we break out of the loop).

(1) Does the `finally` block get called after `except` is done? If so, what does the `break` statement on L213 do? Cause it to try again in the `for attempt in range(45)` loop? Or try once and exit the loop? Maybe I'm getting confused as to whether we're trying to break out of a loop, or `try..except` block